core/src/main/java/com/linecorp/armeria/client/DefaultClientFactory.java

```diff
@@ -171,4 +175,20 @@ public void close() {
     void doClose() {
         clientFactoriesToClose.forEach(ClientFactory::close);
     }
+
+    private URI normalizeUri(URI uri, Scheme scheme) {
+        if (isUndefinedUri(uri)) {
+            // We use a special singleton marker URI for clients that do not explicitly define a
+            // host or scheme at construction time.
+            // As this isn't created by users, we don't need to normalize it.
+            return uri;
+        }
+
+        if (scheme.serializationFormat() != SerializationFormat.NONE) {
+            return uri;
+        }
+
+        return URI.create(scheme.sessionProtocol() +
```

Comment:
`scheme.sessionProtocol().uriText()`?

Reply:
toString() also returns the uriText().

Reply:
Made changes, to explicitly call uriText().

Reply:
Yeah, but it may not be safe to rely on the behavior of `toString()` in the long term. Thanks for making a change!